sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLListener.scala

```diff
@@ -53,7 +53,8 @@ case class SparkListenerSQLExecutionStart(
     physicalPlanDescription: String,
     sparkPlanInfo: SparkPlanInfo,
     time: Long,
-    modifiedConfigs: Map[String, String] = Map.empty)
+    modifiedConfigs: Map[String, String] = Map.empty,
+    jobTags: Set[String] = Set.empty)
```

Comment:
QQ: where will the jobTags in SparkListenerSQLExecutionStart event be used?

Reply:
As a followup to https://github.com/apache/spark/pull/41964, currently SQL Executions associated with a Spark Connect execution are found by using the EXECUTION_ID_KEY of corresponding Spark Jobs.
There are however various Spark Connect executions that start SQL Execution without starting a Spark Job. For example various commands like SHOW TABLES. Currently with https://github.com/apache/spark/pull/41964 the SQL Executions of these will not be linked from the Connect tab. With the jobTag, they can be now identified using the Spark Connect tag. Since this PR is almost ready to merge (hope CI will finish soon), I think using it for the Connect tab could be done in a followup PR?